quizara-backend/routes/user.route.js

```diff
@@ -0,0 +1,84 @@
+const express = require("express");
+const router = express.Router();
+const userSchema = require("../utils/joi.validate.js");
+const { User } = require("../models/user.model.js");
+const bcrypt = require("bcrypt");
+const jwt = require('jsonwebtoken')
+const {verifyToken} = require("../utils/verifyToken")
+
+const env = {
+  TOKEN_SECRET : 'abrakadabra'
+}
+
+
+
+router.route("/signup").post(async (req, res) => {
+  const userDetails = req.body;
+
+  const { error } = await userSchema.validate(userDetails);
+  if (error)
+    return res.json({ success: false, error: error.details[0].message });
+
+  const emailExist = await User.findOne({ email: req.body.email });
+  if (emailExist)
+    return res.json({
+      success: false,
+      error: "Email already exist . Please try again",
+    });
+
+  const salt = await bcrypt.genSalt(10);
```

Comment:
Generating token can be added to `utils` file as well, so that you need to repeat yourself when your project grows